server/ServerFSM.cpp

```diff
@@ -1482,13 +1544,62 @@ sc::result WaitingForMPGameJoiners::react(const JoinGame& msg) {
         }
 
     } else if (client_type == Networking::CLIENT_TYPE_HUMAN_PLAYER) {
+        if (server.IsAuthRequired(player_name)) {
+            // send authentication request
+            player_connection->AwaitPlayer(client_type, client_version_string);
+            player_connection->SendMessage(AuthRequestMessage(player_name, "PLAIN-TEXT"));
+            return discard_event();
+        }
+
         // verify that there is room left for this player
         int already_connected_players = m_expected_ai_player_names.size() + server.m_networking.NumEstablishedPlayers();
         if (already_connected_players >= m_num_expected_players) {
             // too many human players
             ErrorLogger(FSM) << "WaitingForSPGameJoiners.JoinGame : A human player attempted to join the game but there was not enough room.  Terminating connection.";
             server.m_networking.Disconnect(player_connection);
         } else {
+
+            std::string original_player_name = player_name;
+
+            // Remove AI prefix to distinguish Human from AI.
```

Comment:
This is contained in the client_type check on line 1546.
Is this validation that a human player does not have an AI players name?

Reply:
No, it only removes `AI_` prefix from player name if any same it was done in `MPLobby`.